src/plugins/intel_cpu/src/nodes/eye.cpp

```diff
@@ -0,0 +1,142 @@
+    if (getChildEdges().empty())
+        THROW_ERROR << errorPrefix << "has incorrect number of output edges: " << getChildEdges().size();
+}
+
+template<typename T>
+struct Eye::EyeExecute {
+    void operator()(Eye *node) {
+        node->executeSpecified<T>();
+    }
+};
+
+void Eye::execute(dnnl::stream strm) {
+    auto outputPrec = getChildEdgesAtPort(0)[0]->getMemory().getDesc().getPrecision();
+    OV_SWITCH(intel_cpu, EyeExecute, this, outputPrec,
+              OV_CASE(Precision::FP32, float),
+              OV_CASE(Precision::BF16, bfloat16_t),
+              OV_CASE(Precision::I32, int),
+              OV_CASE(Precision::I8, int8_t),
+              OV_CASE(Precision::U8, uint8_t))
+}
+
+void Eye::initSupportedPrimitiveDescriptors() {
+    if (!supportedPrimitiveDescriptors.empty())
+        return;
+    std::vector<PortConfigurator> inDataConf;
+    std::vector<PortConfigurator> outDataConf;
+
+    inDataConf.reserve(inputShapes.size());
+    for (int i = 0; i < inputShapes.size(); ++i)
+        inDataConf.emplace_back(LayoutType::ncsp, Precision::I32);
+    outDataConf.reserve(1);
+    outDataConf.emplace_back(LayoutType::ncsp, convertPrecision(outType));
+
+    addSupportedPrimDesc(inDataConf, outDataConf, impl_desc_type::ref);
+}
+
+template <typename T>
+void Eye::executeSpecified() {
+    const size_t rowNum = getRowNum();
+    const size_t colNum = getColNum();
+    const int shift = getDiagIndex();
+    auto outPtr = getChildEdgeAt(0)->getMemoryPtr();
+    T *dst = reinterpret_cast<T *>(getChildEdgeAt(0)->getMemoryPtr()->GetPtr());
+
+    const size_t batchVolume = getBatchVolume(getBatchShape());
+    const size_t spatialCount = colNum * rowNum;
+    const size_t spatialSize = spatialCount * sizeof(T);
+    const size_t l2CacheSize = dnnl::utils::get_cache_size(2, true);
+    const size_t elementsCount = colNum * rowNum * batchVolume;
+
+    const size_t minSide = std::min(rowNum, colNum);
+    const size_t maxSide = std::max(rowNum, colNum);
+    const size_t absShift = std::abs(shift);
+    const size_t onesPerBatchNum = (absShift <= maxSide - minSide ? minSide :
+                              absShift < maxSide ? minSide - absShift : 0);
+    const size_t dataShift = (shift >= 0 ? shift : -shift * colNum);
+
+    if (spatialSize >= l2CacheSize) {
+        parallel_nt(0, [&](const size_t ithr, const size_t nthr) {
+            size_t start = 0, end = 0;
+            splitter(elementsCount, nthr, ithr, start, end);
+            memset(dst + start, 0, (end - start) * sizeof(T));
+        });
+        for (size_t bShift = 0; bShift < batchVolume * spatialCount; bShift += spatialCount) {
+            parallel_nt(0, [&](const size_t ithr, const size_t nthr) {
+                size_t start = 0, end = 0;
+                splitter(onesPerBatchNum, nthr, ithr, start, end);
+                for (size_t j = start; j < end; j++) {
+                    dst[dataShift + j * (colNum + 1) + bShift] = 1;
```

Comment:
Is there automatic cast to destination type? There is no need to cast like in reference impl: `static_cast<T>(1)`?

Reply:
I've added `static_cast<T>(1)`